### PR TITLE
Preparation for adding block fetch

### PR DESCRIPTION
**Pull request by dcoutts**
Various minor extensions to the `ChainFragment` module and tests.

### CODE REVIEW

File: ouroboros-network/src/Ouroboros/Network/ChainFragment.hs [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeFamilies #-}
```
Comment (coot) [inline ouroboros-network/src/Ouroboros/Network/ChainFragment.hs]: `TypeFamilies`, why are they needed here?

Comment (dcoutts) [reply ouroboros-network/src/Ouroboros/Network/ChainFragment.hs]: https://github.com/input-output-hk/ouroboros-network/pull/394/commits/952d700c67ebfe37d8b6bd6a3927c552bdab94ad#diff-26cb125985373dd021681ddb686ad41eR420`HeaderHash block1 ~ HeaderHash block2`

Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/ChainFragment.hs]: Thanks @dcoutts

**Pull request merged**